text/dependencies/rfcs/0000-dependency-management-overview.md

```diff
@@ -0,0 +1,197 @@
+impossible for non-core development team members to get a full view into how
+the project manages dependencies.
+
+## Background: Current Implementation
+<details>
+<summary> For context, here's how the current implementation works:</summary>
+
+1. Dep-server has a JSON file of `known-versions` for each dependency in GCP.
+2. [Get New Versions
+   Workflow](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/get-new-versions.yml).
+   The dep-server polls for new versions on a timer every hour for every
+   dependency. Polling website URIs are hard coded into the dep-server code.
+   New versions are discovered by taking the difference between the versions
+   the workflow finds and the content of the `known-versions` file.
+3. [Build and Upload
+   Workflow](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/build-upload.yml).
+   The workflow is triggered by the Get New Versions workflow when a new
+   version is discovered. The build process in step 3 is delegated to
+   [cloudfoundry/buildpacks-ci](https://github.com/cloudfoundry/buildpacks-ci/tree/384c051f48fdb4b40521daaacc6afaab87da3796/tasks/build-binary-new)
+   and
+   [cloudfoundry/binary-builder](https://github.com/cloudfoundry/binary-builder/tree/main/recipe)
+   and may involve pulling from source, compiling, or processing the dependency
+   in some way. The code is written in Ruby.  The compiled dependency is
+   uploaded to an S3 bucket. The workflow also gathers metadata (CPEs,
+   licenses, SHA256, etc.) about the dependency using dep-server
+   dependency-specific code.
+4. [Test and Upload
+   Workflow](https://github.com/paketo-buildpacks/dep-server/blob/main/.github/templates/test-upload-metadata.yml).
+   The workflow is triggered by the Build and Upload workflow and receives the
+   metadata from that step. A smoke test is run against the compiled
+   dependency, and if successful, the dependency metadata is uploaded to an AWS
+   S3 bucket.
+5. Endpoint routing for the dep-server is set up through AWS Route 53 and
+   Cloudfront to access dependencies and metadata, but the actual dep-server
+   runs in Google App Engine.
+6. [Update dependencies
+   Workflow](https://github.com/paketo-buildpacks/github-config/blob/main/implementation/.github/workflows/update-dependencies.yml).
+   Each dependency-providing buildpack has a workflow that runs the `jam
+   update-dependencies` command on a timer (or when manually triggered), which
+   will update dependency versions from the dep-server depending on
+   `[[metadata.dependency-constraints]]` listed in the `buildpack.toml`.
+</details>
+
+## Plan Overview
+
+The plan overview provides high level direction for a new dependency management
+approach, split into phases that will help to address the concerns cited in
+the Motivation section of the RFC. Due to the complex nature of the changes,
+the proposals details will be split into separate RFCs.
```

Comment:
Can we approve one of these phases without the others? Seems like we can't approve the overview without approving the phases at least. In which case, I am not sure what gain from splitting the RFC up. I could definitely see us tracking implementation in three phases but maybe it would be better to have the entire plan (including the three phases) in a single document so we can approve it all in one go and reference the details in one place.

Reply:
The phases are pretty dependent on each other, but I think there are some details that are independent. We could agree on Phase 1 (buildpack-related changes) and get that approved, and then go back to the Phase 2 RFC and hash out the details afterwards. The separation of RFCs is tied to the phased implementation, but I do think it makes the content more digestible by separating concerns. It's a lot of content to put into one RFC in my opinion.

Reply:
It makes it easier for me as the RFC author to have a sort of separation of concerns, but there is some overlap that may make it confusing for others. My intention with the phases to make it clearer!